### ISSUE TITLE
Error when reading input file and unit conversion

### ISSUE THREAD

**Issue by alexmaragko** [PAHFIT/pahfit#58]
In the current version of the code the input file format is decided from the input file extension.
If I run pahfit for the example spectra in the data directory it raises an error of "No reader defined for format ...", followed by a table with the available formats.

This is strange because .ipac is included in the available formats, but "M101_Nucleus_irs.ipac" gives the above error.

The same happens with "orion_bar_SWS_with7023shape.txt", although if I rename the file to .ascii it will read the file.

But, it will raise an error in the following line when the unit conversion is happening, because no unit assignment is happening.

So currently we can't run pahfit as: run_pahfit spectrumfile scipack_ExGal_SpitzerIRSSLLL.ipac without modifying the code, and it's harder to test.

**Comment by karllark** [comment]
@alexmaragko : have you installed pahfit?  This is now needed to get the run_pahfit command installed on your system.  Check out the install instructions at:
https://pahfit.readthedocs.io/en/latest/install.html

I think this will solve the problem, but am not sure.  Please try and tell me.  :-)

**Comment by alexmaragko** [comment]
@karllark Yes I have installed pahfit.

**Comment by alexmaragko** [comment]
I will modify the code a bit to get the units in and make some runs for Monday's telecon though. :)

**Comment by karllark** [comment]
Can you post the output error message?  I don't understand why it doesn't work if you have installed it.

**Comment by alexmaragko** [comment]
This is the output for M101_Nucleus_irs.ipac:
Traceback (most recent call last):
  File "/usr/local/bin/run_pahfit", line 4, in <module>
    __import__('pkg_resources').run_script('pahfit==2.0.dev88', 'run_pahfit')
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 750, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 1527, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python3.5/dist-packages/pahfit-2.0.dev88-py3.5.egg/EGG-INFO/scripts/run_pahfit", line 88, in <module>
    pmodel = PAHFITBase(filename=packfile)
  File "/usr/local/lib/python3.5/dist-packages/pahfit-2.0.dev88-py3.5.egg/pahfit/base.py", line 123, in __init__
    param_info = self.read(filename, tformat=tformat)
  File "/usr/local/lib/python3.5/dist-packages/pahfit-2.0.dev88-py3.5.egg/pahfit/base.py", line 373, in read
    t = Table.read(filename, format=tformat)
  File "/usr/local/lib/python3.5/dist-packages/astropy/table/table.py", line 2531, in read
    out = io_registry.read(cls, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/astropy/io/registry.py", line 516, in read
    reader = get_reader(format, cls)
  File "/usr/local/lib/python3.5/dist-packages/astropy/io/registry.py", line 449, in get_reader
    data_format, data_class.__name__, format_table_str))
astropy.io.registry.IORegistryError: No reader defined for format '5/dist-packages/pahfit-2' and class 'Table'.
The available formats are:
           Format           Read Write Auto-identify Deprecated
--------------------------- ---- ----- ------------- ----------
                      ascii  Yes   Yes            No           
               ascii.aastex  Yes   Yes            No           
                ascii.basic  Yes   Yes            No           
                  ascii.cds  Yes    No            No           
     ascii.commented_header  Yes   Yes            No           
                  ascii.csv  Yes   Yes            No           
              ascii.daophot  Yes    No            No           
                 ascii.ecsv  Yes   Yes           Yes           
           ascii.fast_basic  Yes   Yes            No           
ascii.fast_commented_header  Yes   Yes            No           
             ascii.fast_csv  Yes   Yes            No           
       ascii.fast_no_header  Yes   Yes            No           
             ascii.fast_rdb  Yes   Yes            No           
             ascii.fast_tab  Yes   Yes            No           
          ascii.fixed_width  Yes   Yes            No           
ascii.fixed_width_no_header  Yes   Yes            No           
 ascii.fixed_width_two_line  Yes   Yes            No           
                 ascii.html  Yes   Yes           Yes           
                 ascii.ipac  Yes   Yes            No           
                ascii.latex  Yes   Yes           Yes           
            ascii.no_header  Yes   Yes            No           
                  ascii.rdb  Yes   Yes           Yes           
                  ascii.rst  Yes   Yes            No           
           ascii.sextractor  Yes    No            No           
                  ascii.tab  Yes   Yes            No           
                       fits  Yes   Yes           Yes           
                       hdf5  Yes   Yes           Yes           
                    votable  Yes   Yes           Yes           
                     aastex  Yes   Yes            No        Yes
                        cds  Yes    No            No        Yes
                        csv  Yes   Yes           Yes        Yes
                    daophot  Yes    No            No        Yes
                       html  Yes   Yes            No        Yes
                       ipac  Yes   Yes            No        Yes
                      latex  Yes   Yes            No        Yes
                        rdb  Yes   Yes            No        Yes

When I rename orion_bar_SWS_with7023shape.txt to orion_bar_SWS_with7023shape.ascii then the error is for the obs_spectrum['wavelength'] variable in the next line.

Traceback (most recent call last):
  File "/usr/local/bin/run_pahfit", line 4, in <module>
    __import__('pkg_resources').run_script('pahfit==2.0.dev88', 'run_pahfit')
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 750, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 1527, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python3.5/dist-packages/pahfit-2.0.dev88-py3.5.egg/EGG-INFO/scripts/run_pahfit", line 66, in <module>
    obs_x = obs_spectrum['wavelength'].to(u.micron,
  File "/usr/local/lib/python3.5/dist-packages/astropy/table/table.py", line 1222, in __getitem__
    return self.columns[item]
  File "/usr/local/lib/python3.5/dist-packages/astropy/table/table.py", line 106, in __getitem__
    return OrderedDict.__getitem__(self, item)
KeyError: 'wavelength'

**Comment by karllark** [comment]
Can you post what the command you are issuing on the command line is for the 1st part?

For the 2nd part, this is because the names of the columns are different than expected and the units are not defined in the file.  You could add them, but probably need to update the format of the header to be like that for the M101 example.  I did do all this for the example data, just M101.  But clearly should have.

**Comment by alexmaragko** [comment]
run_pahfit M101_Nucleus_irs.ipac scipack_ExGal_SpitzerIRSSLLL.ipac

I have copied the spectrum files into a different directory than pahfit, and making the runs.

**Comment by karllark** [comment]
For the first part, it looks like it is not parsing the string that is the filename correctly.  The format table string it should be passing should be 'ipac' and it looks like it is passing '5/dist-packages/pahfit-2'.  I don't understand.

**Comment by alexmaragko** [comment]
Indeed changing the split index to -1 did the trick. I was able to make runs for both spectra without issues. 

However, the txt extension is not recognized by the astropy reader and probably it should be renamed to ascii, or alternatively set a statement in the code to match txt with ascii extension internally.

Another thing to consider is how to set the units in the ascii input spectrum, so that the unit conversion part of the code applies to everything. I believe only ECSV format can accept units but not the ascii format. For ascii input spectra we could potentially require to have a second comment line with the units, and assign them in the code before the unit conversion.

Let me know what you think before I make a pull request.

**Comment by karllark** [comment]
Another solution is to replace the ascii input spectrum with an ecsv or ipac format version that includes units.

**Comment by karllark** [comment]
Really glad changing the index to -1 worked!

**Comment by karllark** [comment]
You can always make a pull request whenever and then update it as needed (just push the the branch used for the pull request).

**Comment by alexmaragko** [comment]
> Another solution is to replace the ascii input spectrum with an ecsv or ipac format version that includes units.
> I would create a small amount of code that reads in the ascii spectrum, adds units, then outputs as an ecsv file. Would be good to have alternate formats as part of the examples.

You mean create and replace the orion_bar_SWS_with7023shape.txt spectrum with an ipac equivalent to have it permanently in the data folder of pahfit?

**Comment by karllark** [comment]
Yes.

**Comment by alexmaragko** [comment]
Ok will do it first chance.

**Comment by alexmaragko** [comment]
I've turned orion_bar_SWS_with7023shape.txt into ipac format, so now pahfit can read it and fit it.

I've also added another spectrum in the data folder that @ThomasSYLai  sent to me to, and turned it into ecsv format in order to showcase one more input example.

In order to read ecsv files though, the Table.read format must be "ascii.ecsv". But if the input spectrum is e.g., "spectrum.ecsv" it will raise an error. So I've added a line to turn ecsv to ascii.ecsv.

Since we are currently sticking with input formats that have the units defined, ecsv will be the only "ascii"-type format that requires the special handling of ecsv --> ascii.ecsv, so we shouldn't be troubled to specify conversions for other ascii inputs.
